src/maintainer/knowledge_base.rst

```diff
@@ -1050,6 +1050,51 @@ To skip the pypy builds, do the following,
      skip: True         # [python_impl == 'pypy']
 
 
+Using setuptools_scm
+====================
+
+The Python module `setuptools_scm <https://github.com/pypa/setuptools_scm>`_
+can be used to manage a package's version automatically from metadata, such as git tags.
+The package's version string is thus not specified anywhere in the package,
+but encoded in it at install-time.
+
+For conda-build this warrants some attention because the metadata is often not available.
+The are two options to proceed:
+
+*   For Python package also available on PyPi:
+    Use the PyPi tarball as a source, as it will have the metadata encoded
+    (in such a way that ``setuptools_scm`` knows how to find it).
+
+*   Specify the environment variable ``SETUPTOOLS_SCM_PRETEND_VERSION`` with the version string.
+    If specified this environment variable is the principle source for ``setuptools_scm``.
+    There are two ways how to do this:
+
+    -   If you are using build scripts, in ``build.sh`` specify:
+
+        .. code-block:: bash
+
+            export SETUPTOOLS_SCM_PRETEND_VERSION="$PKG_VERSION"
+
+        and in ``bld.bat`` specify:
+
+        .. code-block:: bash
+
+            set SETUPTOOLS_SCM_PRETEND_VERSION=%PKG_VERSION%
+
+        Whereby you use that ``PKG_VERSION`` has been set with the version string,
+        see `Environment variables <https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#env-vars>`_.
+
+    -   Otherwise, if you are directly building from ``meta.yaml``, use for example:
+
+        .. code-block:: yaml
+
+            build:
+              # [...]
+              script:
+                - export SETUPTOOLS_SCM_PRETEND_VERSION="{{ version }}"  # [unix]
+                - set SETUPTOOLS_SCM_PRETEND_VERSION="{{ version }}"  # [win]
+                - "{{ PYTHON }} -m pip install . -vv"
```

Comment:
Based on the comments on mattermost I think this can be:

```suggestion
              script_env:
                - SETUPTOOLS_SCM_PRETEND_VERSION={{version}}
              script: "{{ PYTHON }} -m pip install . -vv"
```

I've opened a PR to document this in conda-build: https://github.com/conda/conda-build/pull/4211

Reply:
That's amazing! Thanks a lot! Let's merge this once the addition to conda-build is done